classes/module/Module.php

```diff
@@ -758,7 +758,11 @@ public function enable($force_all = false)
             }
         }
 
-        if ($this->getOverrides() != null) {
+        if (count($items) == Shop::getTotalShops()) {
+            $force_all = true;
```

Comment:
`$force_all` is parameter of `ModuleCore::enable()`. It should not be modified by it.
Determining which shops are concerned about module enabling and disabling should not be done *inside* `enable()` and `disable()`.
Moreover, you are trying to deflect `$force_all` usage - which concerns enabling/disabling modules - in order to change overrides install/uninstall behaviour.
~I suggest you to have a look at the way overrides are uninstalled and find a way to fix the behaviour there.~
Sadly this is impossible. Overrides are installed / uninstalled for all shops. They can't be ignored for specific shops, and "active" for other ones.

As enabling/disabling a module only affect its behavior, and has nothing to do with overrides, there is no solution to your current issue.